### PR TITLE
[docs] moodleapp/links: Document links clicked in app

### PR DESCRIPTION
**Pull request by dpalou**

### CODE REVIEW

File: data/migratedPages.yml [hunk 1]
```diff
@@ -1234,8 +1234,8 @@ Moodle_App_Customisation:
 - filePath: "/general/app/customisation/index.md"
   slug: "/general/app/customisation"
 Moodle_App_Deep_Linking:
-- filePath: "/general/app/development/deep-linking.md"
-  slug: "/general/app/development/deep-linking"
+- filePath: "/general/app/development/links/deep-linking.md"
+  slug: "/general/app/development/links/deep-linking"
```
Comment (NoelDeMartin) [inline data/migratedPages.yml:1238]:
```suggestion
- filePath: "/general/app/development/link-handling/deep-linking.md"
  slug: "/general/app/development/link-handling/deep-linking"
```

Maybe this is more understandable?

File: general/app/development/development-guide.md [hunk 2]
```diff
@@ -415,5 +415,5 @@ You can learn more about this in the [Acceptance testing for the Moodle App](./t
 
 - [Moodle App Scripts: gulp push](./scripts/gulp-push.md)
 - [Moodle App Accessibility](../accessibility.md)
-- [Moodle App Deep Linking](./deep-linking.md)
+- [Moodle App Deep Linking](./links/deep-linking.md)
```
Comment (NoelDeMartin) [inline general/app/development/development-guide.md:418]:
```suggestion
- [Moodle App Link Handling](./link-handling)
```
I guess we should link to the entire section now. If nothing else, I think the new documentation is more relevant than deep linking so if we only link to one, maybe I'd prefer this one.

File: general/app/development/links/deep-linking.md [hunk 3]
```diff
@@ -91,3 +91,4 @@ moodlemobile://link=https://mysite.es/mod/choice/view.php?id=8
 ## See also
 
 - [Custom URL Scheme Cordova plugin used by the app](https://github.com/EddyVerbruggen/Custom-URL-scheme).
+- [Links in the Moodle App](./links.md)
```
Comment (NoelDeMartin) [inline general/app/development/links/deep-linking.md:94]: Is it necessary to add this, if they are both in the same section? I guess the sidebar should be fairly visible, even in mobile... So yeah I don't know, I wouldn't add it but I guess it's ok if you want to.

File: general/app/development/links/links.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+title: Links in the Moodle App
+sidebar_label: Links
```
Comment (NoelDeMartin) [inline general/app/development/links/links.md:3]:
```suggestion
title: Inner links in the Moodle App
sidebar_label: Inner links
```

I'm not 100% convinced about this suggestion, but I'd like to differentiate this concept from "deep links". Is it inner links vs deep links? Or how else could we refer to them?

File: general/app/development/links/_category_.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+label: Links
```
Comment (NoelDeMartin) [inline general/app/development/links/_category_.yml:1]:
```suggestion
label: Link handling
```

File: general/app/development/links/links.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+sidebar_position: 1
+tags:
+ - Moodle App
+---
+
+## Pressing a link inside the app
+
+When a user presses a link in the Moodle app, the behaviour changes depending on whether the URL is supported by the app or not:
+
+- If the URL belongs to the same site and it's supported by the app, then the app will try to open the corresponding page inside the app. For example if the user presses a link to an assignment then the assignment will be opened inside the app if possible.
```
Comment (NoelDeMartin) [inline general/app/development/links/links.md:13]:
```suggestion
- If the URL belongs to the same site and it's supported by the app, then the app will try to open the corresponding page inside the app. For example if the user presses a link to an assignment, then the assignment will be opened inside the app if possible.
```

File: general/app/development/links/links.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+## Extending the list of supported URLs
+
+The app has a defined list of supported URLs. If you have a plugin adapted to work in the app and you want to support links to your plugin you will need to create a Link Handler. For more information and examples about this, please see the [Link handlers](../plugins-development-guide/index.md#link-handlers) documentation.
+
+## Opening links in an embedded browser
+
+To open a link in an embedded browser instead of an external browser you can use the `data-open-in` attribute:
+
+```html
+<a href="https://domain.com" data-open-in="app">
```
Comment (NoelDeMartin) [inline general/app/development/links/links.md:25]: What are the possible values of `data-open-in`? `app` and `external`? I guess we should document it as well, even if `external` is already the default value.

File: general/app/development/links/links.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+The behaviour can be customised with the following data attributes:
+
+- `data-open-in`: Set it to "app" to open the page in an embedded browser instead of the system browser.
+- `data-app-url-confirm`: A confirmation message to be displayed before opening the browser.
+- `data-app-url-resume-action`: Set it to "refresh" to update the course page when the user goes back to the app. Right now this only works in the course page, but in the future it might be added to other pages.
+
+An example using all the attributes:
+
+```html
+<button data-app-url="https://anotherdomain.com" data-open-in="app" data-app-url-confirm="You need to enrol to the course in the browser." data-app-url-resume-action="refresh"></button>
```
Comment (NoelDeMartin) [inline general/app/development/links/links.md:63]:
```suggestion
<button
    data-app-url="https://anotherdomain.com"
    data-open-in="app"
    data-app-url-confirm="You need to enrol to the course in the browser."
    data-app-url-resume-action="refresh"
></button>
```

File: general/app/development/links/links.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+:::note Notice
+Please notice that students cannot add data attributes to HTML elements when using the Moodle editor, only teachers and users with the right permissions are able to add them.
+:::
+
+## See also
+
+- [Moodle App Deep Linking](./deep-linking.md)
```
Comment (NoelDeMartin) [inline general/app/development/links/links.md:72]:
```suggestion
```

Same as before, I'm not sure if we should be linking those.

**Review by NoelDeMartin — APPROVED** [review]
Here's some suggestions.

File: general/app/development/links/deep-linking.md (hunk 3 above)
Comment (dpalou) [reply general/app/development/links/deep-linking.md]: Right, I removed it.

File: general/app/development/links/links.md (hunk 4 above)
Comment (dpalou) [reply general/app/development/links/links.md]: As we talked in chat, I've used "Links in the app" and "Deep linking" as section titles.

File: general/app/development/links/links.md (hunk 7 above)
Comment (dpalou) [reply general/app/development/links/links.md]: Right now the only possible value is `app`, all other values open the browser.

File: general/app/development/development-guide.md (hunk 2 above)
Comment (dpalou) [reply general/app/development/development-guide.md]: Using the section gave me a broken link error, so now I'm linking to the first page.

**Pull request merged**